packages/phoenix-evals/src/phoenix/evals/models/bedrock.py

```diff
@@ -43,15 +45,18 @@ def _model_name(self) -> str:
 
     def _init_client(self) -> None:
         if not self.client:
-            try:
-                import boto3  # type:ignore
-
-                self.client = boto3.client("bedrock-runtime")
-            except ImportError:
-                self._raise_import_error(
-                    package_name="boto3",
-                    package_min_version=MINIMUM_BOTO_VERSION,
-                )
+            if self.session:
+                self.client = self.session.client("bedrock-runtime")
+            else:
+                try:
+                    import boto3  # type:ignore
+
+                    self.client = boto3.client("bedrock-runtime")
+                except ImportError:
+                    self._raise_import_error(
+                        package_name="boto3",
+                        package_min_version=MINIMUM_BOTO_VERSION,
+                    )
```

Comment:
is this necessary? Just curious. Seems like we now have a ton of paths for creating a client - some of which can be done on the user side.

Reply:
For context if we want to run tests we need to configure the client with (at least) an AWS region name.

I don't think we need to be able to inject a session: we can probably configure the region for CI using environment variables or allow a user to only pass a `region_name` though I think that implies exposing all necessary client configuration variables. However I think being able to inject a session object instead of a client means we don't have to expose things like `region_name` as a kwarg on the `BedrockModel` class and is a bit cleaner and easier to test